### DISCUSSION TITLE
react custom Hook throwing errors !

### DISCUSSION THREAD

**Discussion by mcabappa2019** [vercel/next.js#16761] [Help] [answered]
# Bug report

## Describe the bug
Getting this error when calling a custom hook : 
**Unhandled Runtime Error
TypeError: Invalid attempt to destructure non-iterable instance.
In order to be iterable, non-array objects must have a [Symbol.iterator]() method.**

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

Added screen shots how I got the issue. 

## Expected behavior

Its a simple custom hook and it should work. 

## Screenshots
![image](https://user-images.githubusercontent.com/55290592/91896636-d16ce880-ecb6-11ea-91e0-046c9a0fbcda.png)

![image](https://user-images.githubusercontent.com/55290592/91897140-78ea1b00-ecb7-11ea-9050-77d3d9788497.png)

![image](https://user-images.githubusercontent.com/55290592/91896420-70451500-ecb6-11ea-8bdc-27d9a0aab8ba.png)

## System information

- OS: WINDOWS
- Browser- chrome
- Version of Next.js: 9.4.4
- Version of Node.js: v12.13.0

**Comment by jamesmosier** [comment, marked as answer]
`const [loadCartData] = useLoadCartData(false);`

should be

`const {loadCartData} = useLoadCartData(false);`

since that hook returns as object:

```js
function useLoadCartData(val = false) {
  ...
  return { loadCartData };
}
```

**Reply by mcabappa2019** [reply]
Thanks.
It was a very silly mistake found later.
